### PR TITLE
First proposal to solve issue #438

### PR DESCRIPTION
**Pull request by Martin-Eckleben**
This is the most simple and effective solution I came up with.

I pondered a lot how to do it more "elegant" with a viewchild passed down , a real anchor in the component with an ID passed down etc. etc.
Every solution I came up with had its own drawbacks.

But adding an a to the DOM - clicking it - and removing it on the fly seems to be the by far most encapsulated and easiest way.

My first thoughts were "Couldn´t DOM manipulation be performance heavy in a big page".
But I rejected that thought because one single manual button click that executes it will not have much impact.

My second thought was "Why not create the a once and and save the performance".
But that was countered by "If I add a library to my project and find wild a´s in my DOM I´m annoyed".

So in conclusion I think this is a robust, easy, encapsulated and not too intrusive way of solving the issue.
What do you think?

### CODE REVIEW

**Comment by MurhafSousli** [comment]
Hi @Martin-Eckleben thanks for the PR!
You don't really need to attach the anchor and detach the anchor, anyway the solution looks good, I've already done some changes in the core in the beta version,  I should also support ionic's "window.open" version.

I found this working without attaching / destroying the anchor element.

```ts
_linkElement!: HTMLLinkElement;

this._linkElement = this._renderer.createElement('a');
  this._linkElement.href = finalUrl;

this._linkElement.click();
```

**Comment by Martin-Eckleben** [comment]
Awesome!
Should I adapt it in my PR and you merge it or do you want to adapt it directly and close this PR?

**Comment by MurhafSousli** [comment]
@Martin-Eckleben Your current PR is directed to the master branch, I've already merged the anchor as a default optional share method, Thanks for pointing out the fix!
